dockerfiles/Dockerfile

```diff
@@ -58,9 +59,9 @@ RUN pip3 install --no-cache-dir -r docker.txt
 WORKDIR /usr/src/app/checkouts/
 RUN if [ -n "$GITHUB_TOKEN" ] ; \
         then \
-        git clone --depth 1 https://${GITHUB_TOKEN}@github.com/readthedocs/readthedocs-ext ; \
+        git clone --depth 1 https://${GITHUB_USER}:${GITHUB_TOKEN}@github.com/readthedocs/readthedocs-ext ; \
```

Comment:
Seems these scripts will break with just a TOKEN?

Reply:
Do you mean also checking for `GITHUB_USER` in the `if` above?

Reply:
Is passing the user correct here? Note the following:

Passing the GITHUB_USER and GITHUB_PASSWORD is says this way is not supported anymore (changes introduced in this PR):
```
▶ git clone --depth 1 https://test:badpass@github.com/readthedocs/readthedocs-ext
Cloning into 'readthedocs-ext'...
remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
fatal: Authentication failed for 'https://github.com/readthedocs/readthedocs-ext/'
```

Passing an invalid token, it asks for password:
```
(python-3.8.12)
rtfd/code/readthedocs-corporate  use-new-gh-tokens ✗ 
▶ git clone --depth 1 https://badpass@github.com/readthedocs/readthedocs-ext     
Cloning into 'readthedocs-ext'...
Password for 'https://badpass@github.com': 
Time: 0h:00m:03s                                                                                                                                                                
```

Passing a valid token using the _old way_ (without this PR changes) it works:
```
(python-3.8.12)
rtfd/code/readthedocs-corporate  use-new-gh-tokens ✗
▶ git clone --depth 1 https://${GITHUB_TOKEN}@github.com/readthedocs/readthedocs-ext     
Cloning into 'readthedocs-ext'...
remote: Enumerating objects: 219, done.
remote: Counting objects: 100% (219/219), done.
remote: Compressing objects: 100% (174/174), done.
remote: Total 219 (delta 34), reused 102 (delta 20), pack-reused 0
Receiving objects: 100% (219/219), 307.91 KiB | 748.00 KiB/s, done.
Resolving deltas: 100% (34/34), done.
```

So, I'm not sure these changes are required/valid.

Reply:
Is passing the user correct here? Note the following:

Passing the GITHUB_USER and GITHUB_PASSWORD is says this way is not supported anymore (changes introduced in this PR):
```
▶ git clone --depth 1 https://test:badpass@github.com/readthedocs/readthedocs-ext
Cloning into 'readthedocs-ext'...
remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
fatal: Authentication failed for 'https://github.com/readthedocs/readthedocs-ext/'
```

Passing an invalid token, it asks for password:
```
(python-3.8.12)
rtfd/code/readthedocs-corporate  use-new-gh-tokens ✗ 
▶ git clone --depth 1 https://badpass@github.com/readthedocs/readthedocs-ext     
Cloning into 'readthedocs-ext'...
Password for 'https://badpass@github.com': 
Time: 0h:00m:03s                                                                                                                                                                
```

Passing a valid token using the _old way_ (without this PR changes) it works:
```
(python-3.8.12)
rtfd/code/readthedocs-corporate  use-new-gh-tokens ✗
▶ git clone --depth 1 https://${GITHUB_TOKEN}@github.com/readthedocs/readthedocs-ext     
Cloning into 'readthedocs-ext'...
remote: Enumerating objects: 219, done.
remote: Counting objects: 100% (219/219), done.
remote: Compressing objects: 100% (174/174), done.
remote: Total 219 (delta 34), reused 102 (delta 20), pack-reused 0
Receiving objects: 100% (219/219), 307.91 KiB | 748.00 KiB/s, done.
Resolving deltas: 100% (34/34), done.
```

So, I'm not sure these changes are required/valid.

Reply:
OK, I think something weird happened because, now it works:

```
▶ git clone --depth 1 https://humitos:${GITHUB_TOKEN}@github.com/readthedocs/readthedocs-ext
Cloning into 'readthedocs-ext'...
remote: Enumerating objects: 219, done.
```

So, 🤷🏼